core/secret.go

```diff
@@ -195,7 +206,15 @@ func (store *SecretStore) GetSecretPlaintext(ctx context.Context, idDgst digest.
 	defer store.mu.RUnlock()
 	secret, ok := store.secrets[idDgst]
 	if !ok {
-		return nil, fmt.Errorf("secret %s: %w", idDgst, secrets.ErrNotFound)
+		// It seems like when setting secret using SetSecret, it is stored in the secretStore as Secret@xxh3:......,
+		// but with the new secrets api, the key does not have that prefix.
+		// TODO(rajatjindal): check with Justin/Andrea if we need to handle it differently
```

Comment:
@rajatjindal just checking if this has been validated already :pray:

Reply:
no, this still need validation from @aluzzardi / @jedevc.